File: src/main/java/org/shark/example/controller/mongo/MongoController.java
package org.shark.example.controller.mongo;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.shark.example.datasource.mongo.pojo.ExampleDo;
import org.shark.example.service.mongo.MongoCreateService;
import org.shark.example.service.mongo.MongoReadService;
import org.shark.example.service.pojo.ResponseDto;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RequiredArgsConstructor
@RestController
@RequestMapping("/mongo")
@Slf4j
public class MongoController {

    private final MongoCreateService mongoCreateService;

    @PostMapping
    public ResponseDto<Void> create(@RequestBody ExampleDo exampleDo) {
        return mongoCreateService.start(exampleDo);
    }

    private final MongoReadService mongoReadService;

    @GetMapping
    public ResponseDto<List<ExampleDo>> read() {
        return mongoReadService.start();
    }

}

File: src/main/kotlin/org/shark/example/controller/mongo/MongoController.kt
package org.shark.example.controller.mongo

import org.shark.example.datasource.mongo.pojo.ExampleDo
import org.shark.example.service.mongo.MongoCreateService
import org.shark.example.service.mongo.MongoReadService
import org.shark.example.service.pojo.ResponseDto
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/mongo")
class MongoController(
    private val mongoCreateService: MongoCreateService,
    private val mongoReadService: MongoReadService
) {

    private val log = LoggerFactory.getLogger(MongoController::class.java)

    @PostMapping
    fun create(@RequestBody exampleDo: ExampleDo): ResponseDto<Void> {
        return mongoCreateService.start(exampleDo)
    }

    @GetMapping
    fun read(): ResponseDto<List<ExampleDo>> {
        return mongoReadService.start()
    }
}
